website/docs/docs/configuration.md

```diff
@@ -155,6 +159,46 @@ export default {
 };
 ```
 
+### Custom remote cache provider
+
+You can plug in any remote storage by implementing [`RemoteBuildCache`](https://github.com/callstack/rnef/blob/main/packages/tools/src/lib/build-cache/common.ts#L24) interface. A simplest remote cache provider, that loads artifact from a local directory available on your filesystem, would look like this:
+
+```ts
+class DummyLocalCacheProvider {
+  name = 'dummy';
+  // artifactName is provided by RNEF, and will look like this:
+  // - rnef-android-release-7af554b93cd696ca95308fdebe3a4484001bb7b4
+  // - rnef-ios-simulator-Debug-04c166be56d916367462fc3cb1f067f8ac4c34d4
+  // depending on flags passed to the `run:*` commands
+  async list({ artifactName }) {
+    const url = new URL(`${artifactName}.zip`, import.meta.url);
+    return [{ name: artifactName, url }];
+  }
+  async download({ artifactName }) {
+    const artifacts = await this.list({ artifactName });
+    const filePath = artifacts[0].url.pathname;
+    const fileStream = fs.createReadStream(filePath);
+    return new Response(fileStream);
+  }
+}
+
+const pluginDummyLocalCacheProvider = (options) => () =>
+  new DummyLocalCacheProvider(options);
+```
```

Comment:
let's make this one implement proper class as this will be probably the biggest inspiration for people that want to create their own provider.

Reply:
also this one is missing function for uploading artifacts which is crucial to make it work.